sycl/test-e2e/LLVMIntrinsicLowering/bitreverse.cpp

```diff
@@ -0,0 +1,189 @@
+// Test that llvm.bitreverse is lowered correctly by llvm-spirv
+
+// RUN: %{build} -o %t.O2.out -O2
```

Comment:
Do we really need to pass O2 here?

Reply:
-O2 would not be necessary if we could use __builtin_elementwise_bitreverse for all cases, but there is an issue with this builtin for 8-bit and 16-bit types.